book/src/libraries.md

```diff
@@ -1,7 +1,131 @@
 # Libraries
 
-RuSTy does not currently have support for separate libraries.
-Libraries can be compiled together with the application as a normal files.
+RuSTy does not currently have support for importing _source based libraries_.
+
+Source based libraries can, however, be compiled together with the application as normal files.
+
+Precompiled libraries or system functions can be added using compilation flags or an entry in the `plc.json` file.
+
+System functions can also be added using [External Function](libraries/external_functions.md) for each POU in that library.
+
+## Library Structure
+
+A library is defined by : 
+- A set of `st` interfaces, each interface represents a function that has been precompiled.
+	> In a POU, the interface is the definition and variable section e.g:
+	> ```iecst
+	>	(*Interaface for program example *)
+	>	PROGRAM example
+	>	VAR_INPUT
+	>		a,b,c : DINT
+	>	END_VAR 
+	>	VAR END_VAR
+	> (* End of interface *)
+	>
+	>	(* Implementation *)
+	>	END_PROGRAM
+	> ```
+- A binary file for each architecture the library has been built for (`x86_64-linux-gnu`, `aarch64-linux-gnu`, ..)
+
+
+## Linking libraries using the `rustyc` command line
+
+To include a library when using the `rustyc` command line interface, the include files can be added using the `-i` flag.
+
+Each [POU](pous.md), [Global Variable](variables.md), or [Datatype](datatypes.md) defined in the included files will be added to the project.
+POUs and Global variables included with the `-i` are marked as external, the implementation part of a POU is ignored.
+
+To link the library, two options are then available:
+
+- <a name="shared-libraries">**Shared Libraries** </a>
+
+    A shared library (i.e. extension `.so`) can be linked using the `-l` flag. </br>
+    For a library called `mylib`, when the flag `-lmylib` is passed, the linker will search for a file called `libmylib.so`.
+    > **Note that the `lib<LibName>.so` format is required by the linker for unix like systems**
+
+    The library locations used by the linker are the default search locations of the linker (i.e. `/usr/lib`, `/lib`), additional pathes can be provided using the `-L` flag (e.g `-L/opt/lib` will make the linker also search for files in /opt/lib).</br>
+    Additionally, the environment variable `LD_LIBRARY_PATH` can be defined to append entries to the linker's search location.
```

Comment:
shall we add a link to further documentation for LD_LIBRARY_PATH?  like https://tldp.org/HOWTO/Program-Library-HOWTO/shared-libraries.html

Reply:
good idea, i'll do that